From: Markus Koschany <apo@debian.org>
Date: Mon, 5 Feb 2018 20:32:05 +0100
Subject: gradle

Fix various build failures which are caused by Debian's currently outdated
Gradle version. The patch has to be revised when we ship Gradle >= 4.
---
 build.gradle | 21 +++++++--------------
 1 file changed, 7 insertions(+), 14 deletions(-)

diff --git a/build.gradle b/build.gradle
index 0232026..f72034f 100644
--- a/build.gradle
+++ b/build.gradle
@@ -114,7 +114,7 @@ configure(subprojects.findAll { it.name != 'util' }) {
         options.author = true
         options.version = true
         options.links 'http://docs.oracle.com/javase/7/docs/api/'
-        options.tagletPath single(project(':util').sourceSets.main.output.classesDirs)
+        //options.tagletPath single(project(':util').sourceSets.main.output.classesDirs)
         options.taglets 'ManualTaglet'
         options.taglets 'DochubTaglet'
         options.taglets 'ServerReleaseTaglet'
@@ -156,20 +156,13 @@ configure(subprojects.findAll { it.name != 'util' }) {
 
 def getGitVersion() {
     def outputAsString
-    new ByteArrayOutputStream().withStream { os ->
-        def result = exec {
-            executable 'git'
-            args 'describe', '--tags', '--always', '--dirty'
-            standardOutput = os
-        }
-        outputAsString = os.toString().substring(1).trim()
-    }
+    outputAsString = "3.6.2"
     return outputAsString
 }
 
 configure(subprojects.findAll { it.name != 'util' && it.name != 'mongo-java-driver' }) {
     apply plugin: 'checkstyle'
-    apply plugin: "com.github.spotbugs"
+    //apply plugin: "com.github.spotbugs"
     apply plugin: 'jacoco'
     apply plugin: 'groovy'
     apply plugin: 'codenarc'
@@ -270,23 +263,23 @@ configure(subprojects.findAll { it.name != 'util' && it.name != 'mongo-java-driv
         configProperties.checkstyleConfigDir = configDir
     }
 
-    spotbugs {
+    /*spotbugs {
         excludeFilter = new File(configDir, 'findbugs-exclude.xml')
         sourceSets = [sourceSets.main]
         toolVersion = '3.1.0'
-    }
+    }*/
 
     codenarc {
         toolVersion = '1.0'
         reportFormat = project.buildingWith('xmlReports.enabled') ? 'xml' : 'html'
     }
 
-    tasks.withType(com.github.spotbugs.SpotBugsTask) {
+    /*tasks.withType(com.github.spotbugs.SpotBugsTask) {
         reports {
             xml.enabled = project.buildingWith('xmlReports.enabled')
             html.enabled = !project.buildingWith('xmlReports.enabled')
         }
-    }
+    }*/
 
     tasks.withType(Test) {
         def jdkHome = findProperty("jdkHome")
